### PR TITLE
Feat: Added exampleModes and usageModes settings per sections

### PR DESCRIPTION
**Pull request by rafaesc**

### CODE REVIEW

File: loaders/utils/getSections.js [hunk 1]
```diff
@@ -51,6 +51,14 @@ function processSection(section, config) {
 
 	return {
 		name: section.name,
+		exampleModes:
+			section.exampleModes || config.showCode === undefined
```
Comment (sapegin) [inline loaders/utils/getSections.js]: I think we can use `process` method in the config schema to support deprecated values, and keep them in the schema only. Will be easier to remove later.

File: docs/Configuration.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ By default, Styleguidist will look for `styleguide.config.js` file in your proje
 * [`dangerouslyUpdateWebpackConfig`](#dangerouslyupdatewebpackconfig)
 * [`defaultExample`](#defaultexample)
 * [`editorConfig`](#editorconfig)
+* [`exampleModes`](#examplemodes)
```
Comment (sapegin) [inline docs/Configuration.md]: Shouldn't it be singular — `exampleMode`?

**Comment by viljamis** [comment]
Nice work! @sapegin what do you think it’d require to get this feature out?

**Comment by sapegin** [comment]
It looks good to me! Thanks for all the hard work! I think we can include this in 7.1.0, I’ll try to find some time to make a release.

**Pull request closed without merging**